src/inmanta/agent/cache.py

```diff
@@ -40,13 +40,24 @@ def __init__(self, timeout: int = 24 * 3600, version: int = 0) -> None:
 
 
 class CacheItem:
-    def __init__(self, key: str, scope: Scope, value: Any, call_on_delete: Optional[Callable[[Any], None]]) -> None:
```

Comment:
This while cache evalidation is getting too complicated, I would revert this to what it was and rename `_advance_time` to `clean_stale_entries`  and remove all calls to `_advance_time`